libass/ass_utils.h

```diff
@@ -48,6 +48,8 @@
 
 #define ASS_PI 3.14159265358979323846
 
+#define FEATURE_MASK(feat) (1ULL << (feat))
```

Comment:
Seems a bit odd that all the variables are `uint32_t` but this is an (unsigned) `long long`.

Reply:
The idea was, that we won't need to change the macro in the future; unsigned values gets safely truncated to the range of the variable. `uint32_t` matches the type of the existing header-bitflags.
Would you prefer the macro to use `((uint32_t)1 << (feat))` or the variables to use `unisgned long long`?

Reply:
That was indeed my guess as to the intention, and I can sympathize with it. On the other hand, this may be a little confusing, and it seems a little useless to micro-optimize this one line given that three other lines name the type and therefore would need to be changed anyway.

Ironically, I think I wouldn’t have even nit-picked this if the bit-shift was written inline, but when it’s factored out to a named macro, it somehow feels like it should closely match to the feature-mask variables.

I don’t have too strong of an opinion here. Really, I’d be fine with something even more basic such as `unsigned int`. But if you ask me, `(uint32_t)1` seems like a better option simply because it wastes less memory/registers.

At any rate, feel free to commit/merge any version of this if you want to.

Reply:
`unsigned int` *(at least 16 bits)* may be smaller than `uint32_t`, so `1u << (feat)` doesn't seem like a good idea to me *(`unsigned long long` is at least 64 bits per ISO-C99)*.

Reply:
For one thing, we aren’t even using 16 bits now.

For another, this is your scheduled reminder that libass assumes `int` fits 32 bits. Or do we think all instances have been patched out by MrSmile’s various type-changing commits?

Reply:
> libass assumes int fits 32 bits

But why add new code relying on this?

Anyway, I changed the macro to use `((uitn32_t)1)`. ANother approach would have been to use `unisgned long long` everywhere but the flag variable, so we'd only need to change one place, perhaps when/if we ever need to bump the limit.